server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -1292,6 +1300,7 @@ public ClusterState execute(ClusterState currentState) {
                         shards = snapshotEntry.shards();
                         assert shards.isEmpty();
                         failure = "Snapshot was aborted during initialization";
+                        abortedDuringInit = true;
```

Comment:
This whole step is kind of stupid now in `7.6+` because we don't write anything during `INIT`. Ideally (and I'd do that in a follow-up), we shouldn't move the snapshot to `ABORTED` here but instead just drop it from the cluster state right away and resolve the listener in `beginSnapshot` to not have the redundant CS updates from moving to `ABORTED` and then removing the snapshot from the CS in `beginSnapshot`.

Reply:
The JavaDocs on `beginSnapshot` should be updated as well, as it claims that the snapshot is created in the repo.